### PR TITLE
validate URI scheme in clusterserviceconfig #1136

### PR DESCRIPTION
**Pull request by AbfrmBlr**
Scheme is already validated in XSD. same approach is used here.
URI is already mandatory in `ClusteringServiceConfigurationBuilder`

### CODE REVIEW

File: clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java [hunk 1]
```diff
@@ -40,6 +41,8 @@
     implements ServiceCreationConfiguration<ClusteringService>,
     CacheManagerConfiguration<PersistentCacheManager> {
 
+  private static final String CLUSTER_URI_PATTERN = "terracotta://([^\\]\\[/?#@]+@)?[^:?#/]+(:[1-9][0-9]{0,4})?(/[^\\?#]*)?(\\?[^#]*)?(#.*)?";
```
Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: Why? Why not simply validating the URI `scheme` is `terracotta:`?

Comment (AbfrmBlr) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: oh yes

Comment (albinsuresh) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: Something unrelated to this issue exactly: Do we really want to keep this scheme prefix as `terracotta`? Isn't that a bit too long? `tc://` like old times would have been much better I feel.

Comment (ljacomet) [inline clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java]: `terracotta:` is used all over the place, let's not change that so late - we may open up to a shorter version later.

**Comment by ljacomet** [comment]
Still have `http` in `clustered/client/src/test/resources/configs/resource-sample.xml`

**Comment by AbfrmBlr** [comment]
No tests are failing, this means its not being used. that too it should show error for xsd's

**Comment by ljacomet** [comment]
Indeed - fine with me to delete the file

**Comment by AbfrmBlr** [comment]
test this please

**Pull request merged**